src/Bicep.Core/Semantics/FileSymbol.cs

```diff
@@ -94,6 +111,46 @@ public override void Accept(SymbolVisitor visitor)
 
         public IEnumerable<DeclaredSymbol> GetDeclarationsByName(string name) => this.declarationsByName[name];
 
+        /// <summary>
+        /// Tries to get the semantic module of the Bicep File referenced via a using declaration from the current file.
+        /// If current file is not a parameter file, the method will return false.
+        /// </summary>
+        public bool TryGetBicepFileSemanticModelViaUsing([NotNullWhen(true)] out SemanticModel? semanticModel, [NotNullWhen(false)] out Diagnostic? failureDiagnostic)
+        {
+            semanticModel = null;
+            failureDiagnostic = null;
+            if (this.FileKind == BicepSourceFileKind.BicepFile)
+            {
+                throw new InvalidOperationException($"File '{this.FileUri}' cannot reference another Bicep file via a using declaration.");
+            }
+
+            var usingDeclaration = this.Syntax.Children.OfType<UsingDeclarationSyntax>().FirstOrDefault();
+            if(usingDeclaration is null)
+            {
+                // missing using
+                failureDiagnostic = DiagnosticBuilder.ForPosition(new TextSpan(0, 0)).UsingDeclarationNotSpecified();
```

Comment:
FYI Jonny recently added `DiagnosticBuilder.ForDocumentStart()` for this purpose

Reply:
Neat. I'll switch to it.